docs/datamodel/links.rst

```diff
@@ -475,11 +475,141 @@ in these cases due to better ergonomics of selecting, updating, and even
 casting into :eql:type:`json` when keeping all data in the same place rather
 than spreading it across link and object properties.
 
+
+Inserting and updating link properties
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To add a link with a link property, add the link property to a shape on the
+linked object being added. Be sure to prepend the link property's name with
+``@``.
+
+.. code-block:: edgeql
+
+    insert Person {
+      name := "Bob",
+      family_members := (
+        select detached Person {
+          @relationship := "sister"
+        }
+        filter .name = "Alice"
+      )
+    }
+
+The shape could alternatively be included on an insert if the object being
+linked (the ``Person`` named "Alice" in this example) is being inserted as part
+of the query. If the outer person ("Bob" in the example) already exists and
+only the links need to be added, this can be done in an ``update`` query
+instead of an ``insert`` as shown in the example above.
+
+Updating a link's link property is similar to adding a new one except you can
+select the link on the object instead of selecting from the object type since
+the link has already been established. Here, we've discovered that Alice is
+actually Bob's *step*-sister, so we want to change the link property on the
+already-established link between the two:
+
+.. code-block:: edgeql
+
+    update Person
+    filter .name = "Bob"
+    set {
+      family_members := (
+        select .family_members {
+          @relationship := "step-sister"
+        }
+        filter .name = "Alice"
+      )
+    }
+
+Using ``select .family_members`` here with the shape including the link
+property allows us to modify the existing link's link property.
```

Comment:
Maybe "the link property of the existing link" or just "the property of the existing link"? I guess by definition a property of an existing link is a link property..

Reply:
I doubled up here to try to disambiguate between a link property and a property on the linked object. I guess I probably went overboard. 😅

Reply:
Went with your first suggestion. It's unambiguous without also being ridiculous like my initial attempt. 😆